modtools/intro-loader.lua

```diff
@@ -0,0 +1,160 @@
+-folder
+    Designate a folder to get the input intro files from relative to
+    the DF directory
+    Defaults to /data/announcement/ if not specified
+-now
+    Run the script now, as opposed to waiting for the correct timing
+    For debug use, don't implement in init files.
+-file
+    Change the prefix for input filenames.
+    By default, the prefix is "INTRO", as in "INTRO_MOUNTAIN_1"
+-entity
+    Enables entity-specific intro files
+-rand
+    Enables random selection for intro files
+
+
+Introduction files have to be set up by the user
+Intro files must
+    -Be in the custom format used by Toady, editable via the WTF tools
+     developed by Andux. Available as of now at:
+        http://dffd.bay12games.com/file.php?id=4175
+    -In plaintext, the file must begin with the word "fortressintro" followed
+     by a newline. The first line must be exactly such, or it will fail to be
+     loaded by the game.
+    -The file must be named according to your chosen mode for the script:
+      - Parts of the filename are separated by underscores for readability
+        purposes
+      - Filenames must start with the defined prefix, which defaults to INTRO,
+        but can be changed with the -file parameter
+      - If the -entity argument is supplied, the entity name must be appeneded
+        to the prefix, as in INTRO_MOUNTAIN for dwarves
+      - If the -rand argument is supplied, the filenames must each have a unique
+        number appended, as in INTRO_MOUNTAIN_2
+      - Further, for the -rand argument, usable numbers start at 1 and all
+	    numbers must be consecutive.
+      - The files must not have file extensions
+]====]
+
+
+local utils = require 'utils'
+
+
+validArgs = validArgs or utils.invert({
+    'help',
+    'v'
+    'now',
+    'folder',
+    'file',
+    'rand',
+    'entity',
+})
+
+local args = utils.processArgs({...}, validArgs)
+
+if args.help then
+    print(help)
+    return
+end
+
+local inputfolder = dfhack.getDFPath() .. "\\data\\announcement\\"
+if args.folder then
+    inputfolder = dfhack.getDFPath() ..  args.folder .. "\\"
+end
+
+local prefix = "INTRO"
+if args.file then
+    prefix = args.file
+end
+
+if args.rand then
+    math.randomseed(os.time())
+end
+
+
+--Build current index of files in directory
+do
+    local OS = dfhack.getOSType()
+    if OS == 'windows' then
+        os.execute("dir \"" .. inputfolder .. "\" /L /B >index.txt")
+    elseif OS == 'linux' or OS == 'darwin' then --apple's real name is darwin, apparently
+        os.execute("ls \"" .. inputfolder .. "\" -1 -A >index.txt")
+    end
+    
+    local list = io.open(dfhack.getDFPath() .. "\\data\\announcement\\index.txt","r")
+    indextext = list:read("*all")
+    io.close(list)
+end
+
+
+--Code main body, this is where the magic happens
+local function introloader(folder,dorand,doent,verbose)
+    local entity = "_" .. df.historical_entity.find(df.global.ui.civ_id).entity_raw.code
+    if ~doent then entity = "" end
```

Comment:
This won't work - if `doent` is a non-number (boolean, nil, etc.), this will raise an error. Otherwise, `~0 == -1` and `~1 == -2`, so this will always be true.
(Use `not` instead.)